registration-processor/registration-processor-landing-zone-Utility/src/main/java/io/mosip/registration/processor/util/LandingZoneUtility.java

```diff
@@ -0,0 +1,104 @@
+import java.util.List;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.scheduling.annotation.Scheduled;
+import org.springframework.stereotype.Component;
+
+import io.mosip.commons.khazana.spi.ObjectStoreAdapter;
+import io.mosip.kernel.core.logger.spi.Logger;
+import io.mosip.registration.processor.core.code.ApiName;
+import io.mosip.registration.processor.core.constant.LandingZoneTypeConstant;
+import io.mosip.registration.processor.core.constant.LoggerFileConstant;
+import io.mosip.registration.processor.core.exception.ApisResourceAccessException;
+import io.mosip.registration.processor.core.logger.RegProcessorLogger;
+import io.mosip.registration.processor.core.spi.restclient.RegistrationProcessorRestClientService;
+import io.mosip.registration.processor.status.dto.InternalRegistrationStatusDto;
+import io.mosip.registration.processor.status.dto.RegistrationStatusDto;
+import io.mosip.registration.processor.status.dto.SyncRegistrationDto;
+import io.mosip.registration.processor.status.dto.SyncResponseDto;
+import io.mosip.registration.processor.status.service.RegistrationStatusService;
+import io.mosip.registration.processor.status.service.SyncRegistrationService; 
+@Component
+class LandingZoneUtility{
+	private static Logger regProcLogger = RegProcessorLogger.getLogger(LandingZoneUtility.class);
+	@Value("${landing.zone.account.name}")
+	private String landingZoneAccount;
+		
+	@Value("${landing.zone.type:ObjectStore}")
+	private String landingZoneType;
+	
+	@Value("${registration.processor.packet.ext}")
+    private String extention;
+	
+	@Value("${landing.zone.move.schedule:0 0 0 * * *}")
+    private static  String schedule;
+	
+	 @Autowired
+	 private ObjectStoreAdapter objectStoreAdapter;
+
+	    /**
+	     * The sync registration service.
+	     */
+	 @Autowired
+	 private SyncRegistrationService<SyncResponseDto, SyncRegistrationDto> syncRegistrationService;
+
+	    /**
+	     * The registration status service.
+	     */
+	 @Autowired
+	 private RegistrationStatusService<String, InternalRegistrationStatusDto, RegistrationStatusDto> registrationStatusService;
+
+	 @Autowired
+	 private RegistrationProcessorRestClientService<Object> restClient;
+	 /**
+	  *   
+	  * moves the packets from dmz server to objectstore if the landing zone has been changed to 
+      */
+	 @Scheduled(fixedDelayString = "${mosip.regproc.landing.zone.fixed.delay.millisecs:43200000}",
+	            initialDelayString = "${mosip.regproc.landing.zone.inital.delay.millisecs:300000}")
+	   public void movePacketsToObjectStore() {
+		if(landingZoneType.equalsIgnoreCase(LandingZoneTypeConstant.DMZ_SERVER)) {
+			regProcLogger.debug(LoggerFileConstant.SESSIONID.toString(), LoggerFileConstant.USERID.toString(),
+					"", "PacketUploaderServiceImpl::movePacketsToObjectStore()::entry");
+			List<String> regIdList=registrationStatusService.getAllRegistrationIds();
```

Comment:
There could be millions of packets so list will not even work here. Also I see an issue here. The scheduler will keep uploading packets to objectstore even if its uploaded before. It should be limited by number. In one run only limited number of packets should be processed not all. 
and dont read regidlist from the registration table. Insted use - 
apache fileutils.listFiles to list all files in a directory. Read first few and process them and delete them, Then read next batch like that.
Reference -
http://www.avajava.com/tutorials/lessons/how-do-i-get-all-files-with-certain-extensions-in-a-directory-including-subdirectories.html

Reply:
I have modified the code.